c10/core/TensorImpl.h

```diff
@@ -1503,6 +1508,18 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return is_non_overlapping_and_dense_;
   }
 
+  bool track_view() const {
+    return track_view_;
+  }
+
+  void setup_track_view(bool track_view) {
+    if (!track_view_ && track_view) {
+      TORCH_CHECK(false, "Changing inference_only tensor to track_view=True is not allowd");
+    } 
+    version_counter_ = VariableVersion(track_view ? 0 : -1);
```

Comment:
Aside: should we care about what `version_counter_` might be in the `track_view` true case? I.e., would this ever zero out the version counter when we would want to preserve it?

Not to sidetrack from the main feature decision, just a note for if we go this way.